### PR TITLE
Building a Tiny JS World

### PR DESCRIPTION
**Pull request by ASTkach**
### Tiny JS World

[Demo](https://astkach.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/ASTkach/a-tiny-JS-world/tree/tiny-js)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by ASTkach** [comment]
I have completed self-check.

File: submissions/ASTkach/js-world/index.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+let jsWorld = [dog, cat, woman, man];
+
+let result = jsWorld.map((item) => {
+    return item.species + "; " + 
+           item.name + "; " + 
+           item.gender + "; " + 
+           item.legs + "; " + 
+           item.hands + "; " + 
+           item.saying + "; " + 
+           item.friends;
```
Comment (OleksiyRudenko) [inline submissions/ASTkach/js-world/index.js:59]: Too much repetition of code (`item.`, `+"; " +`).
You will benefit from `Array#map` and `Array#join` to make code dry.

File: submissions/ASTkach/js-world/index.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+});
+
+for (let i = 0; i < result.length; i++) {
+    print(`<strong>${result[i]}</strong>`);
+}
```
Comment (OleksiyRudenko) [inline submissions/ASTkach/js-world/index.js:64]: A classic use case for `Array#forEach`

File: submissions/ASTkach/js-world/index.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    legs: 1,
+    hands: 2,
+    saying: "Honey, where's my wooden leg?",
+    friends: [" Elvis", " Marilyn"],
+};
+
+let jsWorld = [dog, cat, woman, man];
+
+let result = jsWorld.map((item) => {
+    return item.species + "; " +
```
Comment (OleksiyRudenko) [inline submissions/ASTkach/js-world/index.js:53]: When an arrow function body has a single `return` statement then you do not need neither function body `{}` nor `return`

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@ASTkach quite a decent job done!
Let's use contemporary ES6/ES2015 JS features more.

**Comment by OleksiyRudenko** [comment]
The approach to handling objects and their props has changed drastically.
You could take some new learnings by following comments from the mentor, which would result in better code than what the majority of other students do.
You are free not to take the challenge.
Let me know if you want to make a step back and implement improvements based on your previous code or want to stick with current "average" implementation.

**Pull request closed without merging**